src/ui.rs

```diff
@@ -204,4 +292,17 @@ impl Ui {
 
 enum UiOutput {
     Terminal { stdout: Stdout, stderr: Stderr },
+    Paged { child: Child },
+}
+
+impl UiOutput {
+    fn new_paged(settings: &UserSettings) -> UiOutput {
+        let pager_cmd = pager_setting(settings);
+        UiOutput::Paged {
+            child: Command::new(pager_cmd)
+                .stdin(Stdio::piped())
+                .spawn()
+                .expect("Failed to spawn pager"),
```

Comment:
Would be better to return this as an error (`CommandError::UserError`), right?

Reply:
Yes. Doing that makes it clear that this can cause creation of the `Ui` to fail, which is quite troubling to me because that means that we lose our only way of writing to stdout/stderr without breaking our own abstractions. Let me try to find a way that doesn't make `Ui` creation fail.

Reply:
Seems like falling back to unpaged output if the user chose a bad (e.g. non-existent) pager is a good choice

Reply:
To clarify, I suggest that if the call to `UiOutput::new_paged(&settings)` in `Ui::for_terminal()` fails, then we use the unpaged `UiOutput`.

Reply:
Alternatively, maybe we can always create a plain ui and apply user settings later.
`cli_util::create_ui()` has a similar workaround.

```rust
let ui = Ui::new();
// process --color, --pager, --config, etc.
ui.reset(new_settings); // will spawn pager here
```

Anyway, I like the idea of falling back to unpaged output on error.